scripts/generator.go

```diff
@@ -0,0 +1,199 @@
+//go:generate go run generator.go generator_util.go postprocess_funcs.go generate_funcs.go --func=generate --module_name=headersMore
+//go:generate go run generator.go generator_util.go postprocess_funcs.go generate_funcs.go  --func=generate --module_name=njs
+//go:generate go run generator.go generator_util.go postprocess_funcs.go generate_funcs.go --func=generate --module_name=otel
+//go:generate go run generator.go generator_util.go postprocess_funcs.go generate_funcs.go --func=generate --module_name=OSS
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+)
+
+const (
+	ossRepo         = "https://github.com/nginx/nginx.git"
+	ossVersionLimit = 3
+	tmpRootDir      = "./generator_tmp"
+)
+
+const (
+	luaModuleName          = "lua"
+	heardersMoreModuleName = "headersMore"
+	njsModuleName          = "njs"
+	otelModuleName         = "otel"
+	ossName                = "OSS"
+	nPlusName              = "NPLUS"
+)
+
+var module2git = map[string]string{
+	heardersMoreModuleName: "https://github.com/openresty/headers-more-nginx-module.git",
+	luaModuleName:          "https://github.com/openresty/lua-nginx-module.git",
+	njsModuleName:          "https://github.com/nginx/njs.git",
+	otelModuleName:         "https://github.com/nginxinc/nginx-otel.git",
+}
+
+func compare2directiveMap(correct map[string][]uint, generated map[string][]uint) {
+	for directiveName, bitmask := range correct {
+		mBitMask, find := generated[directiveName]
+		if !find {
+			fmt.Println(directiveName + " no found")
+			continue
+		}
+		sort.Slice(mBitMask, func(i, j int) bool {
+			return mBitMask[i] > mBitMask[j]
+		})
+		sort.Slice(bitmask, func(i, j int) bool {
+			return bitmask[i] > bitmask[j]
+		})
+		if len(bitmask) != len(mBitMask) {
+			fmt.Println(directiveName + " no same len")
+			continue
+		}
+		sameV := true
+		humanV := false
+		for idx, v := range mBitMask {
+			if v != bitmask[idx] {
+				sameV = false
+			}
+			if bitmask[idx]&uint(0x00020000) != 0 {
+				humanV = true
+			}
+		}
+		if !sameV && !humanV {
+			fmt.Println(directiveName + " no same v")
+		}
+	}
+}
+
+// todo: delete it
+func testRun() {
+	documentedDirectives, err := fetchDocumentedDirctives()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	fmt.Println(documentedDirectives)
+}
+
+func generateModuleFromWeb(moduleName string) error {
+	repoURL, found := module2git[moduleName]
+	if !found {
+		return fmt.Errorf("can't find git repo for module {%s}, make sure it is in the module2git map (in ./scripts/generator_script.go)", moduleName)
+	}
+
+	moduleTmpDir := path.Join(tmpRootDir, moduleName)
+	if directoryExists(moduleTmpDir) {
+		err := os.RemoveAll(moduleTmpDir)
+		if err != nil {
+			return fmt.Errorf("removing %s failed, please remove this directory mannually", moduleTmpDir)
+		}
+	}
+	defer os.RemoveAll(tmpRootDir)
+
+	err := os.MkdirAll(moduleTmpDir, 0777)
+	if err != nil {
+		return err
+	}
```

Comment:
Don't create a temporary directory this way or worse attempt to obliterate what already exists on disk. Look at [os.CreateTemp](https://pkg.go.dev/os#CreateTemp).

```suggestion
	f, err := os.CreateTemp("", "moduleName")
	if err != nil {
		return err // no need to wrap
	}
	defer os.Remove(f.Name())
```